server/src/main/java/org/opensearch/search/query/QueryCollectorContext.java

```diff
@@ -67,38 +72,69 @@
+            @Override
+            public ScoreMode scoreMode() {
+                return scoreMode;
+            }
+        };
+    }
 
-        @Override
-        Collector create(Collector in) throws IOException {
-            return EMPTY_COLLECTOR;
-        }
+    private static final ReduceableSearchResult EMPTY_RESULT = result -> {};
 
-        @Override
-        CollectorManager<?, ReduceableSearchResult> createManager(CollectorManager<?, ReduceableSearchResult> in) throws IOException {
-            return new CollectorManager<Collector, ReduceableSearchResult>() {
-                @Override
-                public Collector newCollector() throws IOException {
-                    return EMPTY_COLLECTOR;
-                }
+    private static QueryCollectorContext createEmptyContext(String name, Collector collector) {
+        return new QueryCollectorContext(name) {
+            @Override
+            Collector create(Collector in) {
+                return collector;
+            }
 
-                @Override
-                public ReduceableSearchResult reduce(Collection<Collector> collectors) throws IOException {
-                    return result -> {};
-                }
-            };
-        }
-    };
+            @Override
+            CollectorManager<?, ReduceableSearchResult> createManager(CollectorManager<?, ReduceableSearchResult> in) {
+                return new CollectorManager<>() {
+                    @Override
+                    public Collector newCollector() {
+                        return collector;
+                    }
+
+                    @Override
+                    public ReduceableSearchResult reduce(Collection<Collector> collectors) {
+                        return EMPTY_RESULT;
+                    }
+                };
+
+            }
+        };
+    }
+
+    private static final Map<ScoreMode, QueryCollectorContext> CONTEXTS = Arrays.stream(ScoreMode.values())
+        .collect(
+            Collectors.toMap(
+                Function.identity(),
+                scoreMode -> createEmptyContext(formatContextName(scoreMode), createEmptyCollector(scoreMode))
+            )
+        );
+
+    private static String formatContextName(ScoreMode scoreMode) {
+        return String.format(Locale.ROOT, "empty_with_score_mode_%s", scoreMode.toString().toLowerCase(Locale.ROOT));
+    }
+
+    private static final Collector EMPTY_COLLECTOR = createEmptyCollector(ScoreMode.COMPLETE_NO_SCORES);
+    public static final QueryCollectorContext EMPTY_CONTEXT = CONTEXTS.get(ScoreMode.COMPLETE_NO_SCORES);
+
+    /**
+     * Returns the {@link QueryCollectorContext} for the provided {@link ScoreMode}
+     *
+     * @param scoreMode The score mode to get the context for
+     */
+    public static QueryCollectorContext getContextForScoreMode(final ScoreMode scoreMode) {
+        return CONTEXTS.getOrDefault(scoreMode, EMPTY_CONTEXT);
+    }
```

Comment:
Can we add tests for this change?

Reply:
Happy to add tests! Could you suggest which existing test class would be the best fit? I'd prefer not to create a new one just for this method.

Reply:
We can have it in https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/search/query/QueryPhaseTests.java

Reply:
ack

Reply:
added requested tests